Service/ElasticSearchService.php

```diff
@@ -0,0 +1,89 @@
+<?php
+
+namespace Markup\NeedleBundle\Service;
+
+use Doctrine\Common\Collections\ArrayCollection;
+use Elasticsearch\ClientBuilder;
+use function GuzzleHttp\Promise\coroutine;
+use function GuzzleHttp\Promise\promise_for;
+use GuzzleHttp\Promise\PromiseInterface;
+use Markup\NeedleBundle\Adapter\GroupedResultAdapter;
+use Markup\NeedleBundle\Adapter\SolariumResultPromisePagerfantaAdapter;
+use Markup\NeedleBundle\Builder\SolariumSelectQueryBuilder;
+use Markup\NeedleBundle\Context\SearchContextInterface;
+use Markup\NeedleBundle\Query\ResolvedSelectQuery;
+use Markup\NeedleBundle\Query\ResolvedSelectQueryDecoratorInterface;
+use Markup\NeedleBundle\Query\SelectQueryInterface;
+use Markup\NeedleBundle\Result\PagerfantaResultAdapter;
+use Markup\NeedleBundle\Result\SolariumDebugOutputStrategy;
+use Markup\NeedleBundle\Result\SolariumFacetSetsStrategy;
+use Markup\NeedleBundle\Result\SolariumSpellcheckResultStrategy;
+use Pagerfanta\Pagerfanta;
+use Shieldo\SolariumAsyncPlugin;
+use Solarium\Client as SolariumClient;
+use Symfony\Component\Templating\EngineInterface as TemplatingEngine;
+
+/**
+ * A search service using Solr/ Solarium.
+ *
+ * Composes a SelectQuery & SearchContext (into a ResolvedQuery), converts it into a solr query and
+ * executes the query (using Solarium) and then returns a result (as a PagerfantaResultAdapter)
+ */
+class ElasticSearchService // implements SearchServiceInterface
+{
+    /**
+     * Elastic client.
+     *
+     * @var ClientBuilder
+     **/
+    private $elastic;
+
+
+    /**
+     * SolrSearchService constructor.
+     * @param ClientBuilder $elastic
+     */
+    public function __construct(
+        ClientBuilder $elastic
+    )
+    {
+        $this->elastic = new ClientBuilder();
+    }
+
+    public function indexParams($params)
+    {
+        return $this->getElasticQuery()->index($params);
+    }
+
+    public function getDocument($params)
+    {
+        return $this->getElasticQuery()->get($params);
+    }
+
+    public function search($params)
+    {
+//        dump($params);exit;
+        return $this->getElasticQuery()->search($params);
+    }
+
+    public function delete($params)
+    {
+        return $this->getElasticQuery()->delete($params);
+    }
+
+
+    public function deleteIndex($params)
+    {
+        return $this->getElasticQuery()->indices()->delete($params);
+    }
+
+    public function putMapping($params)
+    {
+        return $this->getElasticQuery()->indices()->putMapping($params);
+    }
+
+    private function getElasticQuery()
+    {
+        return $this->elastic::create()->build();
```

Comment:
is there no configuration this? Its unlikely we will have PHP & Elastic search on the same server after all.